src/inspect_ai/tool/_tools/_computer/_resources/tool/computer_tool.py

```diff
@@ -44,29 +53,67 @@ def main():
         sys.exit(1)
 
 
-def parse_arguments():
-    parser = argparse.ArgumentParser(description="Execute computer tool action")
-    parser.add_argument("--action", type=str, required=True, help="Action to perform")
-    parser.add_argument("--text", type=str, help="Optional text parameter")
-    parser.add_argument(
-        "--coordinate",
-        type=int,
-        nargs=2,
-        help="Optional coordinate parameter as a list of two integers",
-    )
-    return parser.parse_args()
-
-
-async def execute_action(args) -> ToolResult:
+async def execute_action(args: Namespace) -> ToolResult:
     # we can't do anything until X11 is ready to go.
     await wait_for_file("/tmp/xfce_started")
 
     computer = X11Client()
-    return await computer(
-        action=args.action,
-        text=args.text,
-        coordinate=args.coordinate if args.coordinate else None,
-    )
+    action: Action = args.action
```

Comment:
This is where all of the `Any` shenanigans are now constrained. We have strictly parsed the command line above, but `argparser` still just returns a bucket of `Any`'s. This code relies on the fact that the parser did its job. All that remains is to make `mypy` happy. That's why you'll see all of the calls to `not_none`.